### PR TITLE
Update dependency-injection.md

### PR DESCRIPTION
**Pull request by kyleburnsdev**
Elaborated on the recommendation to avoid Service Locator pattern

### CODE REVIEW

**Comment by guardrex** [comment]
@kyleburnsdev Take a look at the touch-ups that I made and see what you think.

**Comment by guardrex** [comment]
@scottaddie Ok if I push this forward? These are minor updates, and the PR has been hang'in out for quite a while.

**Comment by kyleburnsdev** [comment]
Updates look good to me

File: aspnetcore/fundamentals/dependency-injection.md [hunk 1]
```diff
@@ -534,17 +534,17 @@ The factory method of single service, such as the second argument to [AddSinglet
...
 
 * Avoid static access to services (for example, statically-typing [IApplicationBuilder.ApplicationServices](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder.applicationservices) for use elsewhere).
 
-* Avoid using the service locator pattern (for example, [IServiceProvider.GetService](/dotnet/api/system.iserviceprovider.getservice)).
+* Avoid using the *service locator pattern*. For example, don't inject an <xref:System.IServiceProvider> and invoke <xref:System.IServiceProvider.GetService*> to obtain a service instance when you can use DI instead. Another service locator variation to avoid is injecting a factory that resolves dependencies at runtime. Both of these practices mix [Inversion of Control](/dotnet/standard/modern-web-apps-azure-architecture/architectural-principles#dependency-inversion) strategies.
```
Comment (pranavkm) [inline aspnetcore/fundamentals/dependency-injection.md]: You don't have to inject a `IServiceProvider` - most containers inject one by default

**Pull request merged**